lib/galaxy/webapps/galaxy/fast_app.py

```diff
@@ -72,6 +77,22 @@ async def add_x_frame_options(request: Request, call_next):
             response.headers['X-Frame-Options'] = x_frame_options
             return response
 
+    nginx_x_accel_redirect_base = getattr(gx_app.config, 'nginx_x_accel_redirect_base', None)
+    apache_xsendfile = getattr(gx_app.config, 'apache_xsendfile', None)
+    if nginx_x_accel_redirect_base or apache_xsendfile:
+
+        @app.middleware("http")
+        async def add_send_file_header(request: Request, call_next) -> Response:
+            response = await call_next(request)
+            if not isinstance(response, FileResponse):
+                return response
+            response = cast(FileResponse, response)
+            if nginx_x_accel_redirect_base:
+                response.headers['X-Accel-Redirect'] = nginx_x_accel_redirect_base + response.path
+            if apache_xsendfile:
+                response.headers['X-Sendfile'] = response.path
```

Comment:
```suggestion
                response.headers['X-Sendfile'] = str(response.path)
```

... was this the first time type-checking prevented a bug in Galaxy ? 😆

Reply:
Awesome! it's never too late :laughing: